python/lsst/daf/butler/formatters/file.py

```diff
@@ -175,9 +175,9 @@ def _coerceBuiltinType(self, inMemoryDataset: Any, writeStorageClass: StorageCla
             and type(inMemoryDataset).__module__ == "builtins"
         ):
             # Try different ways of converting to the required type.
-            if hasattr(writeStorageClass.pytype, "parse_obj"):
+            if hasattr(writeStorageClass.pytype, "model_validate"):
```

Comment:
Hmm, would it be safer to check `isinstance` instead with pydantic base class? `model_validate` is probably not super-popular method name but even that could potentially clash in some random class?

Reply:
I see I'm already using isinstance later so it probably does no harm to say this is explicitly the pydantic path. The best bit about this comment here is that it explains to me why ci_hsc_gen3 failed. I was trying to understand why ScarletModelData was not working properly and it turns out it's because that class tried to emulate pydantic by having a `parse_obj` and `json()` method and so `isinstance` here would break it. It does though tell me how to fix the scarlet problem, or at least I have to decide whether to keep parse_obj here or change `ScarletModelData` to use `model_validate` and `model_dump_json`...